### DISCUSSION TITLE
Review: Allow marking sections of a file as viewed

### DISCUSSION THREAD

**Discussion by iandunn** [community/community#6089] [Pull Requests]
## Problem

When reviewing a file with several new functions, or changes to many functions, I sometimes find it hard to keep track of what I've already reviewed. That's especially true if the review is complex or takes a long time.

This is the same problem that the `Viewed` checkbox solves at the _file_ level. There's still a need for it _within_ files, though.

## Ideas

1. One way could be to allow _functions_ to be marked as viewed. That wouldn't necessarily apply well to things like HTML/CSS, though.
2. Maybe just allow it by Git _hunks_, similar to how `git add -p` lets you stage individual hunks? That wouldn't work when there's long sections of added lines, though, because Git doesn't split those into hunks. I guess GH could add some extra logic on top of Git in order to split things into smaller groups, even if that's as simple as "10 lines" rather than trying to determine a logical split.
3. Maybe let the user highlight a series of consecutive lines, and mark those? The user will know better than any algorithm where they want to split. This could be in addition to any splitting that's easy to automatically do.

Related: #5303, #5054

**Comment by jtzemp** [comment]
Also related:
- https://github.com/orgs/community/discussions/33842
- https://github.com/orgs/community/discussions/22041#discussioncomment-3235305
